src/Algebra/NormalSubgroup.agda

```diff
@@ -0,0 +1,34 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definition of normal subgroups
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+open import Algebra.Bundles using (Group; RawGroup)
+
+module Algebra.NormalSubgroup {c ℓ} (G : Group c ℓ)  where
+
+open Group G
+
+open import Algebra.Structures using (IsGroup)
+open import Algebra.Morphism.Structures
+import Algebra.Morphism.GroupMonomorphism as GM
+open import Data.Product.Base
+open import Level using (suc; _⊔_)
+
+record NormalSubgroup c′ ℓ′ : Set (c ⊔ ℓ ⊔ suc (c′ ⊔ ℓ′)) where
+  field
+    -- N is a subgroup of G
+    N : RawGroup c′ ℓ′
+    ι : RawGroup.Carrier N → Carrier
+    ι-monomorphism : IsGroupMonomorphism N rawGroup ι
+    -- every element of N commutes in G
+    normal : ∀ n g → ∃[ n′ ] g ∙ ι n ∙ g ⁻¹ ≈ ι n′
```

Comment:
So... I was a bit surprised that this was an 'easier' formulation than
```suggestion
    normal : ∀ n g → ∃[ n′ ] g ∙ ι n ≈ ι n′ ∙ g
```
which I wondered as to whether 
* it would be easier/smoother to show gives rise to an equivalence relation on the quotient
* it would generalise (better) to `Loop`, `Quasigroup` or even `Magma`/`Semigroup`?

Cf. comments elsewhere from @JacquesCarette about defining 'ideal' via 'sink'...

Reply:
Elsewhere I think that @JacquesCarette has commented on not (necessarily) letting 'traditional' definitions be our guide, with his citation of 'Post-Modern Algebra' as a rational reconstruction which perhaps avoids the... drawbacks... of 'tradition'.

Reply:
And there's a body of precedent being built where even in Lean, which is unabashedly classical, has often chosen non-traditional definitions because they turned out to *still* be too hard to work with in practice. 'Tradition' is for paper-math; fully formal math needs to carve its own path because it aims higher.